pkg/resource/inference_component/hooks.go

```diff
@@ -0,0 +1,166 @@
+		svcsdk.InferenceComponentStatusDeleting,
+	}
+
+	resourceName = GroupKind.Kind
+
+	lastSpecForUpdateAnnotation = fmt.Sprintf("%s/last-spec-for-update", GroupKind.Group)
+
+	requeueWaitWhileDeleting = ackrequeue.NeededAfter(
+		errors.New(resourceName+" is Deleting."),
+		ackrequeue.DefaultRequeueAfterDuration,
+	)
+)
+
+// customDescribeInferenceComponentSetOutput sets the resource ResourceSynced condition to False if
+// InferenceComponent is being modified by AWS
+func (rm *resourceManager) customDescribeInferenceComponentSetOutput(ko *svcapitypes.InferenceComponent) {
+	latestStatus := ko.Status.InferenceComponentStatus
+	svccommon.SetSyncedCondition(&resource{ko}, latestStatus, &resourceName, &modifyingStatuses)
+}
+
+// customUpdateInferenceComponentSetOutput sets ConditionTypeResourceSynced condition to True or False
+// based on the InferenceComponentStatus on AWS so the reconciler can determine if a
+// requeue is needed
+func (rm *resourceManager) customUpdateInferenceComponentSetOutput(ko *svcapitypes.InferenceComponent) error {
+	//set last inference component spec used for update in annotations
+	annotations := ko.ObjectMeta.GetAnnotations()
+	if annotations == nil {
+		annotations = make(map[string]string)
+	}
+	spec := ko.Spec.DeepCopy()
+	spec.Tags = nil
+	serializedSpec, err := json.Marshal(spec)
+	if err != nil {
+		return err
+	}
+	annotations[lastSpecForUpdateAnnotation] = string(serializedSpec)
+	ko.ObjectMeta.SetAnnotations(annotations)
+
+	// injecting Updating status to keep the Sync condition message and status.InferenceComponentStatus in sync
+	ko.Status.InferenceComponentStatus = aws.String(svcsdk.InferenceComponentStatusUpdating)
+
+	latestStatus := ko.Status.InferenceComponentStatus
+	svccommon.SetSyncedCondition(&resource{ko}, latestStatus, &resourceName, &modifyingStatuses)
+
+	return nil
+}
+
+// customUpdateInferenceComponentPreChecks adds specialized logic to check if controller should
+// proceed with UpdateInferenceComponent call.
+// Update is blocked in the following cases:
+//  1. while InferenceComponentStatus != InService (handled by requeueUntilCanModify method).
+//  2. InferenceComponentStatus == Failed.
+//  3. A previous update to the InferenceComponent with same spec failed.
+//
+// Method returns nil if InferenceComponent can be updated,
+// otherwise InferenceComponentUpdateError depending on above cases.
+func (rm *resourceManager) customUpdateInferenceComponentPreChecks(
+	ctx context.Context,
+	desired *resource,
+	latest *resource,
+	delta *ackcompare.Delta,
+) error {
+	latestStatus := latest.ko.Status.InferenceComponentStatus
+	if latestStatus == nil {
+		return nil
+	}
+
+	failureReason := latest.ko.Status.FailureReason
+
+	desiredSpec := desired.ko.Spec.DeepCopy()
+	desiredSpec.Tags = nil
+
+	var lastSpecForUpdateString *string = nil
+	// get last endpoint config name used for update from annotations
+	annotations := desired.ko.ObjectMeta.GetAnnotations()
+	for k, v := range annotations {
```

Comment:
Do you mean, why loop instead of directly retrieving the value? I can make that change in the next release since functionally there is no change, and this release is time sensitive. Let me know if you think otherwise.